libslide/src/bignum/utils.rs

```diff
@@ -64,6 +97,27 @@ pub fn ifft(item: Vec<Complex>, n: usize) -> Vec<Complex> {
     a
 }
 
+// converts a multi digit polynomial to an single digit vector
+pub fn convert_poly(v: Vec<u16>) -> Vec<u8> {
+    let mut carry: u16 = 0;
+    let mut a: Vec<u16> = v
+        .into_iter()
+        .enumerate()
+        .map(|(_i, mut e)| {
+            e += carry;
+            carry = e / 10;
+            while e / 10 > 0 {
+                e %= 10;
+            }
```

Comment:
By definition this loop will only run once. So why not just `e %= 10`?

Reply:
e can be greater then 100 so it runs more then once

Reply:
Yes, but `e %= 10` will always make `e < 10` right. So then `e/10 > 0` will be false

Reply:
Please fix this